src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java

```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
       byteArrayOutputStream.close();
     }
   }
+
+  @Override
+  public List<BillingForm> filterBillingFormList(List<BillingForm> oldBillingForm,
+      BillingFilter billingFilter) {
+    List<BillingForm> filtered = oldBillingForm
+        .stream()
+        .filter(bf -> {
+          if (billingFilter.getBillingType().equals("all")) {
+            return true;
+          } else if (BillingType.valueOf(billingFilter.getBillingType()) == bf.getBillingType()
+              && BillingType.valueOf(billingFilter.getBillingType()) == BillingType.INCOMING) {
+            return true;
+          } else {
+            return BillingType.valueOf(billingFilter.getBillingType()) == bf.getBillingType()
+                && BillingType.valueOf(billingFilter.getBillingType()) == BillingType.OUTLAY;
+          }
+        })
+        .filter(bf -> bf.getPersonAssigned().getName().equals(billingFilter.getPersonName()))
+        .filter(bf -> bf.isConfirmed() == billingFilter.isConfirmed())
+        .filter(bf -> bf.getDateOfCreation().isAfter(billingFilter.getDateCreationStart()) && bf
+            .getDateOfCreation().isBefore(billingFilter.getDateCreationEnd()))
+        .collect(Collectors.toList());
+    return filtered;
+  }
+
+  @Override
+  public byte[] generateSheetForBillings(List<BillingForm> billingForms) throws IOException {
+    XSSFWorkbook xssfWorkbook = new XSSFWorkbook();
+    XSSFSheet sheet = xssfWorkbook.createSheet("BillingsList");
+    Row row = sheet.createRow(0);
+    headersForSheet(row);
+    int rowCount = 1;
+    for (BillingForm billingForm : billingForms) {
+      row = sheet.createRow(rowCount++);
+      Cell cell = row.createCell(0);
+      cell.setCellValue(billingForm.getTitle());
+      cell = row.createCell(1);
```

Comment:
pls avoid magic numbers like 1,2,3...
create private static final int fields and name it like for example
private static final int NAME_IDX = 1;
private static final int DESCRIPTION_IDX = 2;
and then use it in method code